example/forest-example.cpp

```diff
@@ -1,12 +1,67 @@
 #include <forest/forest.hpp>
+#include <cstdio>
+#include <functional>
 #include <iostream>
+#include <iterator>
 
 constexpr auto foo = forest::literal<64>("<rgb=500><b><i>hello</b></i></rgb> world");
 
+void with_temp_file(std::function<void(std::FILE*)> const fn) {
+	constexpr auto tf_name = "temp.txt";
+
+	auto temp_file = std::fopen(tf_name, "w+");
+
+	if (temp_file == nullptr) { return; }
+
+	fn(temp_file);
+
+	std::fclose(temp_file);
+	temp_file = nullptr;
+	std::remove(tf_name);
+}
+
 int main() {
 	std::cout << foo << '\n';
+
+	std::cout << forest::literal<71>("I <strike>copy</strike> <b>study</b> code from <u>stackoverflow.com</u>") << '\n';
```

Comment:
Might want to reduce the number of magic numbers a bit; eg use `forest::length()` for one or two?

Reply:
How does it differ from say a call to `std::string_view::size`?

Reply:
`forest::length` will compute the size of the string _after_ all replacements have been done. Ideally the library would call this first, figure out the required size, and then return a literal of that size, all at compile time. But getting that to work is largely impossible in C++17. Still, something like this is possible, although not great:
```cpp
constexpr auto text = "<b>hi</b>";
constexpr auto output = forest::literal<forest::length(text)>(text);
```